### ISSUE TITLE
Don't allow `instanceof @Nullable C`

### ISSUE THREAD

**Issue by wmdietl** [typetools/checker-framework#3033]
`instanceof` is only true for a non-null expression https://docs.oracle.com/javase/specs/jls/se13/html/jls-15.html#jls-15.20.2.

Code like this:

```java
if (nbl instanceof @Nullable C)
```

is misleading: the then-branch is not taken for a null value.

A `@Nullable` annotation on the type of an `instanceof` should raise an error.
Having a `@NonNull` annotation is redundant, and could cause a warning.

**Comment by smillst** [comment]
Related issue: https://github.com/typetools/checker-framework/issues/762
